### PR TITLE
[DEV-5823] Fix refresh of authentication state to save auth token for next call

### PR DESCRIPTION
**Pull request by nkassis**

### CODE REVIEW

**Review by pastap — APPROVED** [review]
ok if tests pass :+1: interceptor looks correct

File: src/main/java/com/indico/IndicoClient.java [hunk 1]
```diff
@@ -16,6 +17,8 @@
 import com.indico.mutation.WorkflowSubmission;
 import com.indico.request.GraphQLRequest;
 
+import javax.imageio.IIOException;
```
Comment (goatrocks) [inline src/main/java/com/indico/IndicoClient.java:20]: what's IIOEXception for?

**Review by goatrocks — APPROVED** [review]
had to look at okhttpclient but looks good

File: src/main/java/com/indico/IndicoClient.java (hunk 1 above)
Comment (nkassis) [reply src/main/java/com/indico/IndicoClient.java:20]: lul fixing that damn that is a BAD NAME

**Pull request merged**